src/Common/src/System/Text/SimpleRegex.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Text
+{
+    internal static class SimpleRegex
+    {
+        // Based on wildcmp written by Jack Handy - <A href="mailto:jakkhandy@hotmail.com">jakkhandy@hotmail.com</A>
+        // https://www.codeproject.com/Articles/1088/Wildcard-string-compare-globbing
+
+        /// <summary>
+        /// Perform a match between an input string and a pattern in which the only special character
+        /// is an asterisk, which can map to zero or more of any character in the input.
+        /// </summary>
+        /// <param name="input">The input to match.</param>
+        /// <param name="pattern">The pattern to match against.</param>
+        /// <returns>true if the input matches the pattern; otherwise, false.</returns>
+        public static bool IsMatchWithStarWildcard(ReadOnlySpan<char> input, ReadOnlySpan<char> pattern)
```

Comment:
```suggestion
        public static bool IsMatchWithStarWildcard(string input, string pattern)
```
Saves a consistent 1%...